Controller/AuthorizeController.php

```diff
@@ -100,48 +93,38 @@ class AuthorizeController implements ContainerAwareInterface
     private $templateEngineType;
 
     /**
-     * @var EventDispatcher
+     * @var EventDispatcherInterface
      */
     private $eventDispatcher;
 
-    /**
-     * Sets the container.
-     *
-     * @param ContainerInterface|null $container A ContainerInterface instance or null
-     */
-    public function setContainer(ContainerInterface $container = null)
-    {
-        $this->container = $container;
-    }
-
     /**
      * This controller had been made as a service due to support symfony 4 where all* services are private by default.
-     * Thus, there is considered a bad practice to fetch services directly from container.
-     * @todo This controller could be refactored to do not rely on so many dependencies
+     * Thus, this is considered a bad practice to fetch services directly from container.
+     * @todo This controller could be refactored to not rely on so many dependencies
      *
-     * @param RequestStack           $requestStack
-     * @param SessionInterface       $session
-     * @param Form                   $authorizeForm
-     * @param AuthorizeFormHandler   $authorizeFormHandler
-     * @param OAuth2                 $oAuth2Server
-     * @param EngineInterface        $templating
-     * @param TokenStorageInterface  $tokenStorage
-     * @param UrlGeneratorInterface  $router
-     * @param ClientManagerInterface $clientManager
-     * @param EventDispatcher        $eventDispatcher
-     * @param string                 $templateEngineType
+     * @param RequestStack             $requestStack
+     * @param Form                     $authorizeForm
+     * @param AuthorizeFormHandler     $authorizeFormHandler
+     * @param OAuth2                   $oAuth2Server
+     * @param EngineInterface          $templating
+     * @param TokenStorageInterface    $tokenStorage
+     * @param UrlGeneratorInterface    $router
+     * @param ClientManagerInterface   $clientManager
+     * @param EventDispatcherInterface $eventDispatcher
+     * @param SessionInterface         $session
+     * @param string                   $templateEngineType
      */
     public function __construct(
         RequestStack $requestStack,
-        SessionInterface $session,
         Form $authorizeForm,
         AuthorizeFormHandler $authorizeFormHandler,
         OAuth2 $oAuth2Server,
         EngineInterface $templating,
         TokenStorageInterface $tokenStorage,
         UrlGeneratorInterface $router,
         ClientManagerInterface $clientManager,
-        EventDispatcher $eventDispatcher,
+        EventDispatcherInterface $eventDispatcher,
+        SessionInterface $session = null,
```

Comment:
I think you need to add this at the end

Reply:
It has been introduced after 1.5 so I don't think it's an issue.

Reply:
True, it was introduced to work around services being private by default.